File: scripts/deprecated/session-start-banner.py
#!/usr/bin/env python3
"""
Session Start Banner - Shows project folder prominently and sets terminal title.
Hook: SessionStart
"""

import json
import os
import sys
from pathlib import Path

# Add lib directory to path for imports
lib_dir = Path(__file__).parent / "lib"
if str(lib_dir) not in sys.path:
    sys.path.insert(0, str(lib_dir))

try:
    from debug_logger import log_hook_start, log_hook_end, log_debug
except ImportError:
    # Fallback no-op functions if logger not available
    def log_hook_start(hook, script): pass
    def log_hook_end(hook, script, success=True): pass
    def log_debug(hook, script, msg, level="INFO"): pass

# Force UTF-8 encoding on Windows to avoid UnicodeEncodeError
if hasattr(sys.stdout, "reconfigure"):
    sys.stdout.reconfigure(encoding="utf-8", errors="replace")

def get_project_name():
    """Get current project directory name."""
    if "CLAUDE_PROJECT_DIR" in os.environ:
        return Path(os.environ["CLAUDE_PROJECT_DIR"]).name
    return Path.cwd().name

def get_skill_count():
    """Get count of indexed skills from hybrid registry."""
    try:
        registry_path = Path.home() / ".claude" / "configs" / "hybrid-registry.json"
        if registry_path.exists():
            with open(registry_path, "r", encoding="utf-8") as f:
                data = json.load(f)
                return len(data.get("skills", {}))
    except Exception:
        pass
    return None

def main():
    log_hook_start("SessionStart", "session-start-banner.py")

    try:
        project_name = get_project_name()
        skill_count = get_skill_count()

        log_debug("SessionStart", "session-start-banner.py", f"Project: {project_name}", "INFO")
        log_debug("SessionStart", "session-start-banner.py", f"Skill count: {skill_count}", "INFO")

        # Set terminal title (sticky - survives /compact)
        try:
            sys.stdout.write(f"\033]0;{project_name}\007")
            sys.stdout.flush()
            log_debug("SessionStart", "session-start-banner.py", "Terminal title set", "INFO")
        except:
            pass

        # Print banner (ASCII only for maximum compatibility)
        line = "=" * max(len(project_name) + 4, 40)
        banner_lines = [
            "",
            line,
            f"[>] {project_name}",
            line
        ]

        if skill_count and skill_count > 0:
            banner_lines.append(f"\n[i] {skill_count} skills | /help pour decouvrir")

        log_debug("SessionStart", "session-start-banner.py", f"Banner has {len(banner_lines)} lines", "INFO")
        log_debug("SessionStart", "session-start-banner.py", "Writing banner to stdout", "OUTPUT")

        for line_content in banner_lines:
            print(line_content)

        # Auto-load relevant memories
        load_relevant_memories(project_name)

        print()

        log_hook_end("SessionStart", "session-start-banner.py", success=True)

    except Exception as e:
        log_debug("SessionStart", "session-start-banner.py", f"ERROR: {e}", "ERROR")
        log_hook_end("SessionStart", "session-start-banner.py", success=False)
        # Silent fail - don't break Claude startup
        pass


def load_relevant_memories(project_name: str):
    """Load relevant memories from Claude Mem based on project name."""
    import subprocess

    claude_mem_dir = Path.home() / "OneDrive" / "Coding" / "_Projets de code" / "2025.12 Claude Mem"
    search_script = claude_mem_dir / "search.py"
    python_exe = claude_mem_dir / "venv" / "Scripts" / "python.exe"

    # Check if Claude Mem is installed
    if not search_script.exists() or not python_exe.exists():
        return

    try:
        # Search for memories related to this project
        result = subprocess.run(
            [str(python_exe), str(search_script), project_name],
            capture_output=True,
            text=True,
            timeout=3
        )

        if result.returncode == 0 and result.stdout.strip():
            # Parse output to count memories
            lines = result.stdout.strip().split('\n')
            memory_count = len([l for l in lines if l.strip() and not l.startswith('=')])

            if memory_count > 0:
                print(f"\n📚 {memory_count} session(s) similaire(s) trouvée(s)")
                print("   Use /system:semantic-memory-search for details")
    except (subprocess.TimeoutExpired, Exception):
        # Silent fail - memory loading is optional
        pass

if __name__ == "__main__":
    main()

File: scripts/core/session-start-banner.js
#!/usr/bin/env node
/**
 * Session Start Banner - Display project banner at session start
 * Hook: SessionStart
 *
 * Displays:
 * - Project name
 * - Skill count
 * - Terminal title
 * - Relevant memories from Claude Mem
 */

const fs = require('fs');
const path = require('path');
const os = require('os');
// execSync removed - no longer needed after Claude Mem removal

// Add lib directory to path for imports
const libDir = path.join(__dirname, 'lib');
const debugLogger = require(path.join(libDir, 'debug-logger.js'));
const unifiedLogger = require(path.join(libDir, 'unified-logger.js'));

const { logHookStart: logHookStartOld, logHookEnd: logHookEndOld, logDebug } = debugLogger;

function getProjectName() {
    if (process.env.CLAUDE_PROJECT_DIR) {
        return path.basename(process.env.CLAUDE_PROJECT_DIR);
    }
    return path.basename(process.cwd());
}

function getSkillCount() {
    try {
        const registryPath = path.join(os.homedir(), '.claude', 'configs', 'hybrid-registry.json');
        if (fs.existsSync(registryPath)) {
            const data = JSON.parse(fs.readFileSync(registryPath, 'utf-8'));
            const count = Object.keys(data.skills || {}).length;
            logDebug('SessionStart', 'session-start-banner.js', `Found ${count} skills in registry`, 'INFO');
            return count;
        }
        logDebug('SessionStart', 'session-start-banner.js', 'Registry not found', 'SKIP');
    } catch (e) {
        logDebug('SessionStart', 'session-start-banner.js', `Error reading registry: ${e.message}`, 'ERROR');
    }
    return null;
}

// REMOVED: loadRelevantMemories() function
// This was causing 3-second timeouts on EVERY SessionStart (ETIMEDOUT errors)
// Claude Mem integration will be reimplemented in future with proper async handling
// For now, use /system:semantic-memory-search manually if needed

/* istanbul ignore next */
function main() {
    // Old logging system (keep for compatibility)
    logHookStartOld('SessionStart', 'session-start-banner.js');

    // NEW: Unified logging system
    const startTime = unifiedLogger.logHookStart('SessionStart', 'session-start-banner.js', 'core');

    try {
        const projectName = getProjectName();
        const skillCount = getSkillCount();

        logDebug('SessionStart', 'session-start-banner.js', `Project: ${projectName}`, 'INFO');
        logDebug('SessionStart', 'session-start-banner.js', `Skill count: ${skillCount}`, 'INFO');

        // Set terminal title
        try {
            process.stdout.write(`\x1b]0;${projectName}\x07`);
            logDebug('SessionStart', 'session-start-banner.js', 'Terminal title set', 'INFO');
        } catch (e) {
            logDebug('SessionStart', 'session-start-banner.js', `Error setting title: ${e.message}`, 'ERROR');
        }

        // Print banner
        const lineLength = Math.max(projectName.length + 4, 40);
        const line = '='.repeat(lineLength);
        const bannerLines = [
            '',
            line,
            `[>] ${projectName}`,
            line
        ];

        if (skillCount && skillCount > 0) {
            bannerLines.push(`\n[i] ${skillCount} skills | /help pour decouvrir`);
        }

        logDebug('SessionStart', 'session-start-banner.js', `Banner has ${bannerLines.length} lines`, 'INFO');
        logDebug('SessionStart', 'session-start-banner.js', 'Writing banner', 'OUTPUT');

        bannerLines.forEach(line => console.log(line));

        // REMOVED: loadRelevantMemories() call - was causing 3s timeout
        // See comment above for details

        console.log();

        // Old logging system
        logHookEndOld('SessionStart', 'session-start-banner.js', true);

        // NEW: Unified logging system with duration
        unifiedLogger.logHookEnd('SessionStart', 'session-start-banner.js', startTime, 'success', {
            project_name: projectName,
            skill_count: skillCount
        });

    } catch (e) {
        logDebug('SessionStart', 'session-start-banner.js', `ERROR: ${e.message}`, 'ERROR');
        logHookEndOld('SessionStart', 'session-start-banner.js', false);

        // NEW: Unified logging with error
        unifiedLogger.logHookEnd('SessionStart', 'session-start-banner.js', startTime, 'error', {
            error_message: e.message
        });
    }
}

// Export for testing
/* istanbul ignore else */
if (typeof module !== 'undefined' && module.exports) {
    module.exports = {
        getProjectName,
        getSkillCount
        // loadRelevantMemories removed
    };
}

// Run main if executed directly
/* istanbul ignore if */
if (require.main === module) {
    main();
}
